### PR TITLE
Refresh the demos and the README

### PR DESCRIPTION
**Pull request by notlee**
The primary goal of this PR is to make sure we are showing the right thing in the right way in the registry. The approach taken here is separating and simplifying demos, leaving more advanced topics to the README. Is this the best approach?

- Separates and simplifies demos, making them easier to browse and copy. 
- Favours Github for more advanced documentation, e.g. use of events.
- Updates the readme to be more accurate regarding available data attribute config.
- Adds a few extra tests (I was confirming a feature works as expected).

### CODE REVIEW

File: main.scss [hunk 1]
```diff
@@ -58,7 +58,4 @@ $o-expander-is-silent: true !default;
 	.o-expander__toggle {
 		@include oExpanderToggle;
 	}
-
-	// Don't output twice
-	$o-expander-is-silent: true !global;
```
Comment (JakeChampion) [inline main.scss]: Why was this removed?

Comment (notlee) [reply main.scss]: @JakeChampion My stupidity. 

I noticed it was imported twice in `demos/src/demo.scss` and couldn't figure out why. I assumed it was a mistake and this added to prevent output twice in the demo. 

But I see now why this is helpful for other clients, and that it is included in the  component boilerplate. Will revert. Thanks for catching!
